engineapi/instance_manager.go

```diff
@@ -53,13 +54,25 @@ func GetDeprecatedInstanceManagerBinary(image string) string {
 }
 
 func CheckInstanceManagerCompatibilty(imMinVersion, imVersion int) error {
+	// Skip compatibility check for gPRC proxy implemented version so volume/engine upgrade is not blocked.
+	if imVersion < InstanceManagerProxyMinAPIVersion {
+		return nil
+	}
```

Comment:
Here, we can use 2 kinds of API versions instead:
`CurrentInstanceManagerAPIVersion` stands for the API version of the ProcessManager service/client hence there is no need to bump it. And `CurrentInstanceManagerProxyAPIVersion` would represent the proxy service/client version. 
Then we don't need to skip this compatibility check without blocking the process launching in old IMs.

Reply:
How would we get the version from the IMs? Do you mean to add in a new field in `im.Status.ProxyAPIVersion`?

Reply:
Similar to the existing APIVersion field. Yes. We need to add:
1. The corresponding fields in `im.Status`
2. The corresponding fields in cli and server output

Reply:
Addressed.